pkg/eval/mods/file/file.go

```diff
@@ -0,0 +1,47 @@
+package file
+
+import (
+	"os"
+
+	"src.elv.sh/pkg/eval"
+	"src.elv.sh/pkg/eval/vals"
+)
+
+// elvdoc:fn open
+// ```elvish
+// file:open $filename
+// ```
+// Opens the file for reading.
+// File should be closed after opening using close
+// Example
+//  ```elvish-transcript
+// ~> cat a.txt
+// Some text in file
+// f = (file:open a.txt)
+// ~> cat < $f
+// Some text in file
+// ~> file:close $f
+// ```
```

Comment:
This doc (and the one for `close` below) need more work; you could also have just copied the existing doc from `fopen` and `fclose`. I'll follow up with a commit after merging this.

Reply:
I understand, thank you for suggesting this.